examples/tutorials/analysis-3d/non_detected_source.py

```diff
@@ -0,0 +1,251 @@
+- test the presence of emission from a potential source with given position and morphology and compute integral and differential ULs.
+
+"""
+
+######################################################################
+# Setup
+# -----
+#
+# As usual, let’s start with some general imports…
+#
+
+# %matplotlib inline
+import matplotlib.pyplot as plt
+
+import astropy.units as u
+
+from gammapy.datasets import MapDataset, Datasets
+from gammapy.estimators import FluxPointsEstimator, ExcessMapEstimator
+from gammapy.modeling import Fit, select_nested_models
+from gammapy.modeling.models import (
+    PointSpatialModel,
+    PowerLawSpectralModel,
+    SkyModel,
+    create_crab_spectral_model,
+)
+
+
+######################################################################
+# Check setup
+# -----------
+#
+
+from gammapy.utils.check import check_tutorials_setup
+
+check_tutorials_setup()
+
+######################################################################
+# Load observation
+# -----------
+#
+# For computational purposes, we have
+# already created a `~gammapy.datasets.MapDataset` from observation id ``20275`` from the
+# public H.E.S.S. data release and stored it in ``$GAMMAPY_DATA``
+#
+
+dataset = MapDataset.read("$GAMMAPY_DATA/datasets/empty-dl4/empty-dl4.fits.gz")
+dataset.peek()
+plt.show()
+
+######################################################################
+# Create Upper Limit maps
+# -----------------------
+#
+# We will first use the `~gammapy.estimators.ExcessMapEstimator` for a quick check to see if
+# there are any potential sources in the field. You may also use the
+# `~gammapy.estimators.TSMapEstimator`.
+
+estimator = ExcessMapEstimator(
+    sum_over_energy_groups=True,
+    selection_optional="all",
+    correlate_off=True,
+    correlation_radius=0.1 * u.deg,
+)
+
+lima_maps = estimator.run(dataset)
+
+significance_map = lima_maps["sqrt_ts"]
+excess_map = lima_maps["npred_excess"]
+
+# We can plot the excess and significance maps
+fig, (ax1, ax2) = plt.subplots(
+    figsize=(11, 4), subplot_kw={"projection": lima_maps.geom.wcs}, ncols=2
+)
+ax1.set_title("Significance map")
+significance_map.plot(ax=ax1, add_cbar=True)
+ax2.set_title("Excess map")
+excess_map.plot(ax=ax2, add_cbar=True)
+plt.show()
+
+######################################################################
+# The significance map looks rather flat! You can plot a histogram of the
+# significance distribution to confirm that it is a standard Gaussian.
+# Deviations from a standard normal can suggest the presence of gamma-ray sources,
+# or can also originate from incorrect modeling of the residual hadronic background.
+# We can also see the correlated upper limits at any position in the map. But note that
+# this is **not** a source UL, as the containment correction is not applied here.
+#
+
+lima_maps.flux_ul.plot(add_cbar=True, cmap="viridis")
+plt.show()
+
+
```

Comment:
Maybe one can then precise that this is not the recommended way to compute UL... The following code is the one.